### ISSUE TITLE
Element query executed before Craft is fully initialized

### ISSUE THREAD

**Issue by jrrdnx** [solspace/craft3-freeform#1190]
Sentry is continuing to report `Element query executed before Craft is fully initialized.` warnings, as previously reported in #1019, #720, and #582.

We're seeing this on multiple projects running Freeform 5.0.14.1, 5.1.7, and 5.1.10.

Stack trace:
```
#0 /home/user/domain.com/vendor/yiisoft/yii2/db/QueryBuilder.php(227): craft\elements\db\ElementQuery->prepare()
#1 /home/user/domain.com/vendor/yiisoft/yii2/db/Query.php(157): yii\db\QueryBuilder->build()
#2 /home/user/domain.com/vendor/yiisoft/yii2/db/Query.php(287): yii\db\Query->createCommand()
#3 /home/user/domain.com/vendor/craftcms/cms/src/db/Query.php(275): yii\db\Query->one()
#4 /home/user/domain.com/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1616): craft\db\Query->one()
#5 /home/user/domain.com/vendor/craftcms/cms/src/elements/User.php(546): craft\elements\db\ElementQuery->one()
#6 /home/user/domain.com/vendor/yiisoft/yii2/web/User.php(698): craft\elements\User::findIdentity()
#7 /home/user/domain.com/vendor/craftcms/cms/src/web/User.php(500): yii\web\User->renewAuthStatus()
#8 /home/user/domain.com/vendor/yiisoft/yii2/web/User.php(199): craft\web\User->renewAuthStatus()
#9 /home/user/domain.com/vendor/yiisoft/yii2/web/User.php(377): yii\web\User->getIdentity()
#10 /home/user/domain.com/vendor/yiisoft/yii2/base/Component.php(139): yii\web\User->getId()
#11 /home/user/domain.com/vendor/solspace/craft-freeform/packages/plugin/src/Bundles/Form/SpamControl/Checks/AbstractCheck.php(26): yii\base\Component->__get()
#12 Solspace\Freeform\Bundles\Form\SpamControl\Checks\AbstractCheck->__construct()
#13 /home/user/domain.com/vendor/yiisoft/yii2/di/Container.php(411): ReflectionClass->newInstanceArgs()
#14 /home/user/domain.com/vendor/yiisoft/yii2/di/Container.php(186): yii\di\Container->build()
#15 /home/user/domain.com/vendor/solspace/craft-freeform/packages/plugin/src/Library/Bundles/BundleLoader.php(50): yii\di\Container->get()
#16 /home/user/domain.com/vendor/solspace/craft-freeform/packages/plugin/src/Freeform.php(723): Solspace\Freeform\Library\Bundles\BundleLoader::loadBundles()
#17 /home/user/domain.com/vendor/solspace/craft-freeform/packages/plugin/src/Freeform.php(233): Solspace\Freeform\Freeform->initBundles()
#18 /home/user/domain.com/vendor/yiisoft/yii2/base/BaseObject.php(109): Solspace\Freeform\Freeform->init()
#19 /home/user/domain.com/vendor/yiisoft/yii2/base/Module.php(161): yii\base\BaseObject->__construct()
#20 /home/user/domain.com/vendor/craftcms/cms/src/base/Plugin.php(122): yii\base\Module->__construct()
#21 craft\base\Plugin->__construct()
#22 /home/user/domain.com/vendor/yiisoft/yii2/di/Container.php(419): ReflectionClass->newInstanceArgs()
#23 /home/user/domain.com/vendor/yiisoft/yii2/di/Container.php(170): yii\di\Container->build()
#24 /home/user/domain.com/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get()
#25 /home/user/domain.com/vendor/craftcms/cms/src/Craft.php(59): yii\BaseYii::createObject()
#26 /home/user/domain.com/vendor/craftcms/cms/src/services/Plugins.php(943): Craft::createObject()
#27 /home/user/domain.com/vendor/craftcms/cms/src/services/Plugins.php(228): craft\services\Plugins->createPlugin()
#28 /home/user/domain.com/vendor/craftcms/cms/src/base/ApplicationTrait.php(1649): craft\services\Plugins->loadPlugins()
#29 /home/user/domain.com/vendor/craftcms/cms/src/web/Application.php(111): craft\web\Application->_postInit()
#30 /home/user/domain.com/vendor/yiisoft/yii2/base/BaseObject.php(109): craft\web\Application->init()
#31 /home/user/domain.com/vendor/yiisoft/yii2/base/Application.php(204): yii\base\BaseObject->__construct()
#32 yii\base\Application->__construct()
#33 /home/user/domain.com/vendor/yiisoft/yii2/di/Container.php(419): ReflectionClass->newInstanceArgs()
#34 /home/user/domain.com/vendor/yiisoft/yii2/di/Container.php(170): yii\di\Container->build()
#35 /home/user/domain.com/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get()
#36 /home/user/domain.com/vendor/craftcms/cms/src/Craft.php(59): yii\BaseYii::createObject()
#37 /home/user/domain.com/vendor/craftcms/cms/bootstrap/bootstrap.php(250): Craft::createObject()
#38 /home/user/domain.com/vendor/craftcms/cms/bootstrap/web.php(40): require()
#39 /home/user/domain.com/web/index.php(14): require()
```

**Comment by kjmartens** [comment]
Sorry for the trouble, @jrrdnx. We'll check into this issue shortly. 🙂

**Comment by kjmartens** [comment]
This should be resolved in Freeform 5.1.11 now. 🙂

**Comment by olivierbon** [comment]
Seeing this in v4 also, can you apply on there too, please?
